components/logins/src/logins.udl

```diff
@@ -2,32 +2,72 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-namespace logins { };
+namespace logins {
+    // We expose the crypto primitives on the namespace
 
+    // Create a new, random, encryption key.
+    [Throws=LoginsStorageError]
+    string create_key();
 
-// Converted from mozilla.appservices.logins.protobuf.rs
-dictionary Login {
-    string id;
+    // Decrypt a string into an EncryptedFields struct. This will be used to get the credentials
+    // from a `Login` dict.
+    [Throws=LoginsStorageError]
+    EncryptedFields decrypt_fields([ByRef]string ciphertext, [ByRef]string encryption_key);
```

Comment:
actually, this can probably die as `Login` has a method that does the same thing.

Reply:
well, it does in rust, but it's a dictionary in UDL, so this needs to remain.